In [1]:
import numpy as np

In [4]:
fpath = r'C:\Users\user\Downloads\Sample_data.csv'

In [5]:
x= np.genfromtxt(fpath,delimiter =',', skip_header=1)
print(x)

[[ 0. 25. 40.]
 [ 1. 30. 35.]
 [ 2. 28. 38.]
 [ 3. 32. 41.]
 [ 4. 26. 37.]
 [ 5. 29. 36.]
 [ 6. 27. 39.]
 [ 7. 33. 42.]
 [ 8. 31. 38.]
 [ 9. 28. 37.]
 [10. 30. 40.]
 [11. 29. 36.]]


# Measures of Central Tendency:


### 1. Calculate the mean sales for Product A and Product B. Compare the average monthly sales for the two products and discuss any observed trends.

In [22]:
Product_A = x[:,1]
Product_B = x[:,2]

a = np.mean(Product_A)
b = np.mean(Product_B)

print('Mean Sale for Product A : ', a)
print('Mean Sale for Product B : ', b)

Mean Sale for Product A :  29.0
Mean Sale for Product B :  38.25


* The average monthly sales for Product A is approximately 29.0
* The average monthly sales for Product B is 38.25

### 2. Find the median sales for each product and discuss how it reflects the central tendency of the sales distribution.

In [21]:
Product_A = x[:,1]
Product_B = x[:,2]

a = np.median(Product_A)
b = np.median(Product_B)

print('Median Sale for Product A : ', a)
print('Median Sale for Product B : ', b)

Median Sale for Product A :  29.0
Median Sale for Product B :  38.0


**Central Tendency:** Both medians (29 for Product A and 38 for Product B) serve as robust indicators of central tendency. They are less sensitive to extreme values compared to the mean. The medians suggest that, in general, half of the sales values fall below the indicated values, providing a sense of the typical sales level for each product.

### 3. Identify the mode of the sales data for both Product A and Product B. Discuss the significance of the mode in the context of these products.

In [40]:
import statistics

Product_A = x[:,1]
Product_B = x[:,2]

a = statistics.mode(Product_A)
b = statistics.mode(Product_B)

print('Mode Sale for Product A : ', a)
print('Mode Sale for Product B : ', b)

Mode Sale for Product A :  30.0
Mode Sale for Product B :  40.0


### 4. Calculate the weighted mean of the sales data, assuming each month has an equal weight. Discuss any implications of using weighted mean in this context.

In [46]:
Product_A = x[:,1]
Product_B = x[:,2]

weight=np.ones(12)

weighted_A = weight * Product_A
a = np.mean(weighted_A)

weighted_B = weight * Product_B
b = np.mean(weighted_B)

print('Weighted mean of the sales data of Product A :', a)
print('Weighted mean of the sales data of Product B :', b)

Weighted mean of the sales data of Product A : 29.0
Weighted mean of the sales data of Product B : 38.25


# Measures of Dispersion:

### 1. Compute the range of monthly sales for both Product A and Product B. Discuss how the range provides insights into the variability of sales.

In [26]:
range_A = x[:,1].max() - x[:,1].min()
range_B = x[:,2].max() - x[:,2].min()

print(range_A)
print(range_B)

8.0
7.0


### 2. Calculate the variance of the sales data for both products. Discuss what the variance reveals about the spread of sales for each product.


![](https://www.onlinemathlearning.com/image-files/population-variance.png)

In [30]:
Product_A = x[:,1]
Product_B = x[:,2]


a = np.mean(Product_A)
variance_product_a = np.sum((Product_A - a)**2) / len(Product_A)

b = np.mean(Product_B)
variance_product_b = np.sum((Product_B - b)**2) / len(Product_B)

print("Variance of Sales for Product A:", variance_product_a)
print("Variance of Sales for Product B:", variance_product_b)

Variance of Sales for Product A: 5.166666666666667
Variance of Sales for Product B: 4.354166666666667


* A higher variance for Product A would indicate a greater spread or dispersion in monthly sales values.This could suggest that sales values for Product A vary more widely around the mean, indicating potential fluctuations or variability in demand.
* A lower variance for Product B would indicate less spread or dispersion in monthly sales values.This could suggest more consistent sales performance, with sales values clustering closely around the mean.
* High variance may be associated with higher uncertainty and volatility in sales. Low variance may suggest more predictable and stable sales patterns.

### 3. Determine the standard deviation of the sales for Product A and Product B. Discuss the significance of standard deviation in understanding the variability in sales.

![](https://media.geeksforgeeks.org/wp-content/uploads/20230605181802/Standard-Deviation-Formula.png)

In [31]:
Product_A = x[:,1]
Product_B = x[:,2]

a = np.mean(Product_A)
standard_deviation_product_a = np.sqrt(np.sum((Product_A - a)**2) / len(Product_A))

b = np.mean(Product_B)
standard_deviation_product_b = np.sqrt(np.sum((Product_B - b)**2) / (len(Product_B)))

print("Standard Deviation of Sales for Product A:", standard_deviation_product_a)
print("Standard Deviation of Sales for Product B:", standard_deviation_product_b)

Standard Deviation of Sales for Product A: 2.273030282830976
Standard Deviation of Sales for Product B: 2.0866640042581523


# Relative Measures of Dispersion:

### 1 . Calculate the coefficient of variation for the sales data of Product A. Discuss how this relative measure of dispersion helps in comparing the variability of products with different average sales.

In [36]:
Product_A = x[:,1]

a = np.mean(Product_A)

Std = np.std(Product_A)

coeff=(Std/a)*100

print('Coefficient of variation for the sales data of Product A :',coeff)

Coefficient of variation for the sales data of Product A : 7.838035458037848


### 2. Compute the relative range for Product B. Discuss how the relative range accounts for the scale of the sales data.

In [37]:
Product_B = x[:,2]

B = np.max(Product_B) - np.min(Product_B)

b = np.mean(Product_B)

relative_range_B = (B/b)*100

relative_range_B

18.30065359477124

* product B is mean is 38.25 and relative range of product B = 18.30065359477124
* A product with a high mean sales but a low relative range has more stable sales than a product with a low mean sales but a high relative range

### 3. Determine the relative standard deviation for Product A. Interpret the result and discuss its implications for understanding the variability relative to the mean.

In [39]:
Product_A = x[:,1]

a = np.mean(Product_A)

Std = np.std(Product_A)

rel_std = (Std / a)*100

print('relative standard deviation for Product A :',rel_std)

relative standard deviation for Product A : 7.838035458037848


Product A mean is 29.0 and relative standard deviation of product A is 7.838035458037848, 
Product A is with low CV values have less relative variability around their mean, sales are stable consistent